sdk/core/Azure.Core/src/Internal/SequenceWriterAccess.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Serialization;
+
+namespace Azure.Core.Internal
+{
+    internal sealed class SequenceWriterAccess
+    {
+        private readonly IModelJsonSerializable<object> _model;
+        private readonly ModelSerializerOptions _options;
+
+        private SequenceWriter? _writer;
```

Comment:
Should we rename this type to PooledSequence or something like that? I am looking at the usage sources and this type is treated as a writer in one place in the codebase and as a set of pooled buffers in hundreds.